MSVC warning C4068 "unknown pragma"

Output warning message in Visual Studio 2017: `warning C4068: unknown pragma`.

Visual Studio Compiler does not support `#pragma GCC ...`.

Please check the compiler before `#pragma GCC`.

--- a/include/pqxx/internal/ignore-deprecated-post.hxx
+++ b/include/pqxx/internal/ignore-deprecated-post.hxx
@@ -1,2 +1,6 @@
 /// End a code block started by "ignore-deprecated-pre.hxx".
+#if defined(__GNUC__)
+
 #pragma GCC diagnostic pop
+
+#endif // __GNUC__

--- a/include/pqxx/internal/ignore-deprecated-pre.hxx
+++ b/include/pqxx/internal/ignore-deprecated-pre.hxx
@@ -11,5 +11,9 @@
  * "ignore-deprecated-post.hxx".  To avoid mistakes, keep the enclosed area as
  * small as possible.
  */
+#if defined(__GNUC__)
+
 #pragma GCC diagnostic push
 #pragma GCC diagnostic ignored "-Wdeprecated-declarations"
+
+#endif // __GNUC__

